### PR TITLE
Use correct RedisSerializer

### LINKED ISSUE #309: Hardcoded JdkSerializationRedisSerializer in RedisOperationsSessionRepository
Spring Session 1.1.0 M1

The following code should likely use the `defaultRedisSerializer` supplied to the `RedisHttpSessionConfiguration`:

```java
	@SuppressWarnings("unchecked")
	public void onMessage(Message message, byte[] pattern) {

... snip

		if(channel.startsWith(getSessionCreatedChannelPrefix())) {
			RedisSerializer<Object> serializer = new JdkSerializationRedisSerializer();
			Map<Object,Object> loaded = (Map<Object, Object>) serializer.deserialize(message.getBody());
			handleCreated(loaded, channel);
			return;
		}

... snip

	}
```

This code works fine if I do not supply my own `RedisSerializer`, but if I follow the instructions in the [release announcement][1] and supply my own `RedisSerializer` (one backed by Kryo in this case) then I get the following exception when the `RedisOperationsSessionRepository` tries to process the session created event message:

```java
2015-11-23T14:22:04,946 WARN org.springframework.data.redis.listener.RedisMessageListenerContainer or.sp.da.re.li.RedisMessageListenerContainer.287 [redisMessageListenerContainer-20] Execution of JMS message listener failed, and no ErrorHandler has been set.
org.springframework.data.redis.serializer.SerializationException: Cannot deserialize; nested exception is org.springframework.core.serializer.support.SerializationFailedException: Failed to deserialize payload. Is the byte array a result of corresponding serialization for DefaultDeserializer?; nested exception is java.io.StreamCorruptedException: invalid stream header: 01006A61
        at org.springframework.data.redis.serializer.JdkSerializationRedisSerializer.deserialize(JdkSerializationRedisSerializer.java:41) ~[spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        at org.springframework.session.data.redis.RedisOperationsSessionRepository.onMessage(RedisOperationsSessionRepository.java:450) ~[spring-session-1.1.0.M1.jar:?]
        at org.springframework.data.redis.listener.RedisMessageListenerContainer.executeListener(RedisMessageListenerContainer.java:245) [spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        at org.springframework.data.redis.listener.RedisMessageListenerContainer.processMessage(RedisMessageListenerContainer.java:235) [spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        at org.springframework.data.redis.listener.RedisMessageListenerContainer$1.run(RedisMessageListenerContainer.java:960) [spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_66]
Caused by: org.springframework.core.serializer.support.SerializationFailedException: Failed to deserialize payload. Is the byte array a result of corresponding serialization for DefaultDeserializer?; nested exception is java.io.StreamCorruptedException: invalid stream header: 01006A61
        at org.springframework.core.serializer.support.DeserializingConverter.convert(DeserializingConverter.java:78) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.core.serializer.support.DeserializingConverter.convert(DeserializingConverter.java:36) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.data.redis.serializer.JdkSerializationRedisSerializer.deserialize(JdkSerializationRedisSerializer.java:39) ~[spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        ... 5 more
Caused by: java.io.StreamCorruptedException: invalid stream header: 01006A61
        at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:806) ~[?:1.8.0_66]
        at java.io.ObjectInputStream.<init>(ObjectInputStream.java:299) ~[?:1.8.0_66]
        at org.springframework.core.ConfigurableObjectInputStream.<init>(ConfigurableObjectInputStream.java:64) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.core.ConfigurableObjectInputStream.<init>(ConfigurableObjectInputStream.java:50) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.core.serializer.DefaultDeserializer.deserialize(DefaultDeserializer.java:68) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.core.serializer.support.DeserializingConverter.convert(DeserializingConverter.java:73) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.core.serializer.support.DeserializingConverter.convert(DeserializingConverter.java:36) ~[spring-core-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.data.redis.serializer.JdkSerializationRedisSerializer.deserialize(JdkSerializationRedisSerializer.java:39) ~[spring-data-redis-1.6.1.RELEASE.jar:1.6.1.RELEASE]
        ... 5 more
``` 

I believe this is because the body of the message received has been serialized with the custom `RedisSerializer` supplied.

[1]: https://spring.io/blog/2015/11/17/spring-session-1-1-0-m1-released

### PR DESCRIPTION
**Pull request by jvalkeal**
- Remove hardcoded JdkSerializationRedisSerializer from onMessage
  in favour of same defaultRedisSerializer user configured as this
  is most likely the one which works.
- Fixes #309

### CODE REVIEW

File: spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java [hunk 1]
```diff
@@ -446,8 +459,8 @@ public void onMessage(Message message, byte[] pattern) {
 
 
 		if(channel.startsWith(getSessionCreatedChannelPrefix())) {
-			RedisSerializer<Object> serializer = new JdkSerializationRedisSerializer();
-			Map<Object,Object> loaded = (Map<Object, Object>) serializer.deserialize(message.getBody());
+			// TODO: is this thread safe?
+			Map<Object,Object> loaded = (Map<Object, Object>) defaultSerializer.deserialize(message.getBody());
```
Comment (jvalkeal) [inline spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java]: Actually I think this casting is wrong and was related to use of jdk serialization. Been playing with kryo and I'm now getting casting errors. Need to play with it a bit more.

**Comment by jvalkeal** [comment]
closing to come up with better impl.

**Comment by jvalkeal** [comment]
It looks like trouble was in my kryo classes, but this PR doesn't make sense unless we actually provide some sensible custom kryo implementation for serializer. That'd be to actually verify things in tests.

File: spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java (hunk 1 above)
Comment (rwinch) [inline spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java]: @jvalkeal What issue were you having exactly?

**Comment by rwinch** [comment]
We spoke offline and the concern was more related to Spring Session scoped beans in general and not this fix. I merged this into master at fe17e3fcc0988ae52c8a50d0007017c595f3769a I also added a test cc1d3a7e9ef85e68835800dd5fdfdeacf66c0444

**Pull request closed without merging**